### PR TITLE
Container Type Content Update and Register Container Type API

### PR DESCRIPTION
**Pull request by cindylay**
## Category

- [x] Content fix
- [x] New article
 

## What's in this Pull Request?
This PR contains updates to Container Type article and includes new API documentation for registering a container type. The `.yml`  file is updated to show the new article in the table of contents

Updates to:
Content updates to `containertypes.md`
Content updates to `auth.md`
Content updates to `cta.md`
TOC addition to `toc.yml`

New article addition:
`register-api-documentation.md`

### CODE REVIEW

**Comment by cindylay** [comment]
I'd like to only merge the 4 mentioned files from the description:
`containertypes.md`
`auth.md`
`toc.yml`
`register-api-documentation.md`

Is there a way to remove the other changed files from this PR? @andrewconnell @VesaJuvonen

File: docs/embedded/concepts/app-concepts/auth.md [hunk 1]
```diff
@@ -24,6 +24,58 @@ SharePoint Embedded supports both App-Only and Delegated (App+User) calls for en
 
 Both App-Only and Delegated SharePoint Embedded from trusted (or private) client applications are allowed. SharePoint Embedded calls to create containers, either App-Only or Delegated, from public clients are blocked by SharePoint Embedded authorization management.
 
+## Container.Selected Scope 
+
+The [Register Container Type API](register-api-documentation.md) API uses the Container.Selected scope. You will to call this API. To use it, you must configure this scope in your App manifest.  
+
+>>  Note that other SharePoint Embedded Graph APIs run with the FileStorageContainer.Selected Scope on Microsoft Graph
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/auth.md:31]: - this should use the MSDOCS alert format
- `Scope` should be lowercase
- should the scope be bold?

File: docs/embedded/concepts/app-concepts/auth.md [hunk 2]
```diff
@@ -24,6 +24,58 @@ SharePoint Embedded supports both App-Only and Delegated (App+User) calls for en
...
+The [Register Container Type API](register-api-documentation.md) API uses the Container.Selected scope. You will to call this API. To use it, you must configure this scope in your App manifest.  
+
+>>  Note that other SharePoint Embedded Graph APIs run with the FileStorageContainer.Selected Scope on Microsoft Graph 
+
+
+### Configure your App Manifest
+
+In your app manifest you will need to: 
+
+Select Manage > Manifest from the left-hand navigation. Locate the property requiredResourceAccess and edit it so it looks like the following JSON:
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/auth.md:38]: markdown: 
- steps in navigation should be bold (Manage > Manifest)
- reqiredResourceAccess should be formatted as inline code

File: docs/embedded/concepts/app-concepts/auth.md [hunk 3]
```diff
@@ -24,6 +24,58 @@ SharePoint Embedded supports both App-Only and Delegated (App+User) calls for en
...
+  }
+],
+```
+
+## Required Permissions
+|      ScopeName     |     Type    |                                                                                                                                Description                                                                                                                                |
+|:------------------:|:-----------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| Container.Selected | Application | Allows the application to utilize the file storage container platform to manage containers without a signed in user. The specific file storage containers and the permissions granted to them will be configured in Microsoft 365 by the developer of each container type. |
+
+> You must configure the `Container.Selected` scope in your App manifest
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/auth.md:76]: Should this be marked with a MSDOC alert, such as IMPORTANT?

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 4]
```diff
@@ -1,47 +1,92 @@
...
 # SharePoint Embedded Container Types
 
-In SharePoint Embedded, all files and documents are stored in Containers, and each Container is identified by a Container Type.
+In SharePoint Embedded, files and documents are stored in Containers, each identified by a Container Type, which is a property stamped on every instance. A Container Type links all Containers to a specific Azure Subscription for billing. Each Container Type is owned by a single SharePoint Embedded Application, and each SharePoint Embedded Application can own only one Container Type. It specifies the access permissions an SharePoint Embedded Application has for all Containers of that type, such as creating, reading, writing, deleting containers, and managing container permissions.
 
-Container Type is a property stamped on every Container instance. Each Container Type is owned by one Application; and each Application can own only one Container Type.
+## Two Kinds of Container Types
+There are two kinds of Container Types that a developer tenant can create:
+1. [Trial Container Types](#trial-container-types)
+2. [Standard Container Types](#standard-container-types)
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:15]: Markdown
- numbered bullets should always be `1.`... don't use explicit numbers... let the rendering engine take care of that
- headings, paragraphs, and bullet lists should have line breast around them (there should be empty lines after lines 12 & 13)

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 5]
```diff
@@ -1,47 +1,92 @@
...
+ 
+## Trial Container Types
+To enable customers to explore the development of SharePoint Embedded applications and assess its features, we offer the ability to create a Trial Container Type. Each customer can have only one Trial Container Type in their tenant at a time. Trial Container Types are not linked to a billing profile and can only be utilized by the tenant developing the application; in this scenario, the partner tenant is always the same as the consuming tenant. The Trial Container Type remains valid for up to 30 days but can be removed at any time within this period. There are two ways to create a Trial Container Type:
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:18]: Please see earlier comment about line breaks for headings & paragraphs

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 6]
```diff
@@ -1,47 +1,92 @@
...
+> The Trial Container Type is the only Container Type that is restricted to the Partner (creating) tenant and cannot be shared with other consuming tenants.
+> 
+> It is not possible to convert a Trial Container Type and/or the associated Containers to a Standard Container Type.
 
-The permission mapping between a ContainerTypeID-AppID pair determines the operations the application is authorized to invoke against all containers of a specific Container Type and their content.
 
-> [!TIP]
-> To learn more about application architecture, see [Application Architecture](./app-architecture.md).
+## Standard Container Types
+Standard Container Types are associated with a billing profile in [Azure Portal](portal.azure.com) when they are created. The billing profile includes a n Azure Subscription ID and a Region Group. There are two pre-requisites to creating s Standard Container Type
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:39]: Please see earlier comment about line breaks for headings & paragraphs

Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:39]: Invalid URL... missing https://

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 7]
```diff
@@ -1,47 +1,92 @@
...
+- An Azure subscription and Resource Group must be present in the [Azure Portal](portal.azure.com).
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:40]: Invalid URL, missing https://

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 8]
```diff
@@ -1,47 +1,92 @@
...
+- An App regisration must be created in [Azure Entra]
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:41]: It's "Microsoft Entra" + missing link

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 9]
```diff
@@ -1,47 +1,92 @@
...
-- [`Remove-SPOContainer`](/powershell/module/sharepoint-online/remove-spocontainer)
-- [`Remove-SPODeletedContainer`](/powershell/module/sharepoint-online/remove-spodeletedcontainer)
-- Remove-SPOContainerType
+## Viewing ContainerTypes
+Developer Admin can view all the SharePoint Embedded Container Types created by them on their tenant using this PowerShell commandlet. This cmdlet retrieves and returns the list of Container Types  created for a SharePoint Embedded Application in the tenant. 
+
+```ps
+Get-SPOContainerType   
+```
+Example of an output of the Get-SPOContainerType  cmdlet
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:67]: Please see earlier comment about line breaks for paragraphs

File: docs/embedded/concepts/app-concepts/containertypes.md [hunk 10]
```diff
@@ -1,47 +1,92 @@
...
+ContainerTypeName   : ContosoLegal
+OwningApplicationId : a735e4af-b86e-0000-93ba-1faded6c39e1
+Classification      : Standard
+AzureSubscriptionId : 564e9025-f7f5-xxx9-9ddd-4cdxxxx1755
+ResourceGroup       : prod-resources
+Region              : EastUS
+```
+
+## Registering Container Types
+To register both Trial and Standard Container Types, please refer to the following [tutorial](../app-concepts/register-api-documentation.md)
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/containertypes.md:79]: Please see earlier comment about line breaks for headings & paragraphs

File: docs/embedded/concepts/app-concepts/register-api-documentation.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+
+As the registration API controls the permissions that a SharePoint Embedded application can perform against the Container in the Consuming Tenant, this should be one of the first APIs invoked. Failure to do so will result in access denied errors when invoking other APIs against Container and/or the content in the Containers.
+
+There are no restrictions on how many times the registration API can be invoked. How often the registration API is invoked and when it is invoked is dependent on the SharePoint Embedded application. Just be aware  that the last successful invoke and associated settings will be the settings persisted and hence used in the Consuming Tenant.
+
+> Only the Owning Application of the Container Type can invoke the registration API in the Consuming Tenant.
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/register-api-documentation.md:16]: Should this be a MSDOC alert, like "NOTE"?

File: docs/embedded/concepts/app-concepts/register-api-documentation.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+
+## HTTP request
+
+ 
+``` http
+PUT {RootSiteUrl}/_api/v2.1/storageContainerTypes/{containerTypeId}/applicationPermissions
+```
+
+> NOTE that this is NOT a Graph API
```
Comment (andrewconnell) [inline docs/embedded/concepts/app-concepts/register-api-documentation.md:51]: Should be MSDOC NOTE format

File: docs/embedded/tutorials/create-containertype.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
+---
+title: Create a Standard Container Types
+description: This article explains how to create a Container Type.
+ms.date: 11/28/2023
+ms.localizationpriority: high
+---
+
+<!--
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:8]: Remove TODO comments from final doc submission

File: docs/embedded/tutorials/create-containertype.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+
+# Create a Standard Container Type in SharePoint Embedded
+SharePoint Embedded [stores](../concepts/app-concepts/app-architecture.md) content securely in File Storage Containers. Each Container has a Container Type. No Containers can be created until a Container Type is created.  
+
+The [Container Type](../concepts/app-concepts/containertypes.md), represented by a `ContainerTypeId` asssociates all Containers linked to it with a specific Azure Subscription. This creates a billing relationship between the SharePoint Embedded application and Containers of a Container Type. This article will explain how to create a Standard [Container Type](../concepts/app-concepts/containertypes.md). Read more about Container Types [here](../concepts/app-concepts/containertypes.md). If you would like to try out SharePoint Embedded for free you can quickly get started with a [Trial Container Type](https://learn.microsoft.com/en-us/sharepoint/dev/embedded/concepts/app-concepts/containertypes#sharepoint-embedded-trial-container-types).
+
+
+**Pre-requisites:**
+* You must have admin credentials to a M365 tenant
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:27]: Please see earlier comment about line breaks for headings & paragraphs

File: docs/embedded/tutorials/create-containertype.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+
+
+
+## Set up an Azure Subscription Id, Resource Group, and Region
+
+### 1. Set up Azure Subscription Id
+Login to Azure Portal with the admin credentials of your tenant
+
+
+(TODO: add the extra existing Sub id)
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:38]: incomplete

File: docs/embedded/tutorials/create-containertype.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+**Pre-requisites:**
+* You must have admin credentials to a M365 tenant
+* You must register your application in [Microsoft Entra ID](https://learn.microsoft.com/en-us/training/modules/sharepoint-embedded-setup/5-exercise-setup-configure-sharepoint-embedded). See tutorial on how to [register an app with Microsoft Entra ID](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/walkthrough-register-app-azure-active-directory).
+
+
+
+## Set up an Azure Subscription Id, Resource Group, and Region
+
+### 1. Set up Azure Subscription Id
+Login to Azure Portal with the admin credentials of your tenant
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:35]: missing period

File: docs/embedded/tutorials/create-containertype.md [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+
+
+(TODO: add the extra existing Sub id)
+
+Navigate to [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBladeV2)
+![alt text](./regct-image/azuresubscription.png)
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:41]: Image alt text must be unique & should be meaningful

File: docs/embedded/tutorials/create-containertype.md [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+![alt text](./regct-image/resource-group.png)
+
+Click "Create" to create a resource group and name it something appropiate to your application. Then click Review + Create
+![alt text](./regct-image/c-rg.png)
+
+Make sure that:
+* The appropiate Azure Subcription is selected for the resource group
+* The appropiate physical region is selected. For ex. (US) East US
+
+>> Note: The Region Group does NOT require your Containers to live in that specified region. Your content is not restricted to the specified region
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:59]: Should use MSDOCS "NOTE" alert"

File: docs/embedded/tutorials/create-containertype.md [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+
+Region refers to where the CT is created  for example “US-East”. This does NOT affect geographically where containers can live and does not affect geographically where applications are installed. Also, regions do not place limits on where SharePoint Embedded Applications can be used.
+
+Click Review + Create and your validation should pass if the inputs are correct.
+
+Now you have all the necessary resources to create a Standard Container Type which will defines the billing relationship between your Containers and Azure SubscriptionId.
+
+<br></br>
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:67]: Don't use HTML in markdown files... this should be removed.

File: docs/embedded/tutorials/create-containertype.md [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+
+## Create your Container Type
+
+### 3. Use PowerShell to create a new Container Type for your Application
+To learn more about the role of Container Types in SharePoint Embedded please refer to the [ContainerTypes](../concepts/app-concepts/containertypes.md) and [App Archictecture](../concepts/app-concepts/app-architecture.md) articles. To learn more about trial containers which are limited, but don't incur costs during the development of your application, see [Developer Admin](../concepts/admin-exp/dev-admin.md).
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:72]: Please see earlier comment about line breaks for headings & paragraphs

File: docs/embedded/tutorials/create-containertype.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+
+You need to install and import the latest **Microsoft.Online.SharePoint.PowerShell** module.
+
+```powershell
+Install-Module "Microsoft.Online.SharePoint.PowerShell" # if you don't have it already
+# OR
+Update-Module "Microsoft.Online.SharePoint.PowerShell" # if you have it already
+```
+
+Have the following information handy from [Azure Portal](portal.azure.com):
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:82]: Invalid link (see earlier comment).

Please see earlier comment about line breaks for headings & paragraphs

File: docs/embedded/tutorials/create-containertype.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+> [!TIP]
+> To learn more about trial containers which are limited, but don't incur costs during the development of your application, see [Developer Admin](../concepts/admin-exp/dev-admin.md).
+
+The cmdlet outputs your new `ContainerTypeId`, save this for later use.
+
+### 3. Grant admin consent for your app
+
+TODO: Take this out of the article and put it into the register CT article....
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:104]: incomplete

File: docs/embedded/tutorials/create-containertype.md [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+### Calling the Register Container Type API
+
+See the API documentation for Register File Storage Container Type Application Permissions [here](register-api.md)
+* Please Note that you will need to use a cert-based access token to make this API call.
+  
+
+### Using the SharePoint Embedded Postman Collection
+
+Use the [Postman collection](https://github.com/microsoft/SharePoint-Embedded-Samples/tree/main/Postman) from the [SharePoint Embedded Samples repo](https://github.com/microsoft/SharePoint-Embedded-Samples) to register the new Container Type on your tenant. 
+* Please Note: This API call must be done using Application-only auth.
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:129]: MSDOC Alert

File: docs/embedded/tutorials/create-containertype.md [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+## Register Your Container Type in the Consuming Tenant
+
+### 4. Register the new Container Type
+
+There are two options to register your new Container Type and provide application permissions.
+
+### Calling the Register Container Type API
+
+See the API documentation for Register File Storage Container Type Application Permissions [here](register-api.md)
+* Please Note that you will need to use a cert-based access token to make this API call.
```
Comment (andrewconnell) [inline docs/embedded/tutorials/create-containertype.md:123]: MSDOC alert

**Review by andrewconnell — CHANGES_REQUESTED** [review]
This isn't a complete submission... there are lots of "TODO" comments throughout the doc. If you're working on the content, please submit the PR as a DRAFT. Open PR's indicate it's ready for review & merge (see [this](https://github.blog/2019-02-14-introducing-draft-pull-requests/) & [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request)).

This PR seems to overlap the other one you submitted recently #9551... is that necessary anymore? 

Strongly suggest you run Acrolinx against each of the MD files you're creating to find & sometimes automatically fix many of the issues I've called out. [Learn more here](https://review.learn.microsoft.com/en-us/help/wwl/publish-content/acrolinx?branch=main). It checks the content for grammar, typos, and issues that violate MSDOCS content rules.

In addition to my comments, review all issues reported in the automated build and fix them as unless there's an exception, we can't merge them until fixed. It's reporting a lot of broken links in the content submission.

**Comment by andrewconnell** [comment]
@cindylay said:
> I'd like to only merge the 4 mentioned files from the description: `containertypes.md` `auth.md` `toc.yml` `register-api-documentation.md`

That's not possible... we can't edit (include/exclude files) what's in your PR. You need to submit the PR from a branch with only the changes you want to make. See https://github.com/SharePoint/sp-dev-docs/wiki/How-to-Create-Good-Pull-Requests

**Comment by andrewconnell** [comment]
@cindylay can you review the findings in the automated build... those are pointing to broken anchor tags and the update to the TOC doesn't appear invalid - https://github.com/SharePoint/sp-dev-docs/pull/9562#issuecomment-1984856717

**Review by arakeshmicrosoft — APPROVED** [review]
We have customers asking for this document. I have reviewed and it provides all the neccessary information to get our customers started.

**Review by tonchan-msft — APPROVED** [review]

**Review by andrewconnell — APPROVED** [review]

**Pull request merged**